### ISSUE TITLE
1.8 Feedback: Allow description in the block for a field

### ISSUE THREAD

**Issue by esjee** [rmosolgo/graphql-ruby#1158]
It appears that in 1.8 it's no longer possible to provide a description using the block argument. Instead the only way to provide a description is to have it be the third argument. So:

Pre 1.8:
```
# this works
   field :triggered_pre_submission_trigger, -> { TriggerType } do
    description 'A pre-submission trigger that notified the hacker before submission. '\
                'This field is only present for reports filed after February 14, 2016.'
  end
```

1.8:
```
# this works
   field(
     :triggered_pre_submission_trigger,
     -> { TriggerType },
     'A pre-submission trigger that notified the hacker before submission. '\
       'This field is only present for reports filed after February 14, 2016.'
     null: true
```

1.8:
```
# this doesn't work, but I would like it to
   field(
     :triggered_pre_submission_trigger,
     -> { TriggerType },
     null: true do
     description 'A pre-submission trigger that notified the hacker before submission. '\
       'This field is only present for reports filed after February 14, 2016.'      
      end
```

There are two reasons why I would like the support the latter.

- It makes it easier to create an automatic upgrade (see graphql_upgrader) to 1.8
- This is subjective, but I think the second style is a bit nicer when it comes to long descriptions

**Comment by rmosolgo** [comment]
It's ok with me to support it, I heard somewhere (?) that it was not helpful to have "so many ways to do things", and better to have one way. So, either I picked the _wrong way_, or we should keep having the different ways!

**Comment by esjee** [comment]
In general, I think that comment makes sense. I can see how it would be helpful to have one 'official' way of doing things. It makes the library less complex. 

At the same time though, this usage can already be observed in classess. For example:

1.8
```
# This works
class Types::TodoList < GraphQL::Schema::Object # or Scalar, Enum, Union, whatever
  description "Things to do (may have already been done)"
end
```

For this reason, I feel that the added complexity is not a major concern.

Are there any other reasons I could have missed to warrant not implementing this?

**Comment by rmosolgo** [comment]
🤷‍♂️  Unless someone disagrees, go for it!

**Comment by swalkinshaw** [comment]
Semi-related, but it kind of surprised me that `description` wasn't a kwarg in 1.8; it's just a positional argument. It's weird since descriptions are optional but I guess since it's the *last* positional arg it matters less.

**Comment by rmosolgo** [comment]
Yeah, I think a kwarg would be nice too, it makes for some weird, long lines as it is.

**Comment by rmosolgo** [comment]
Fixed by #1175
